addons/survey/static/tests/tours/survey_tour_session_manage.js

```diff
@@ -31,47 +34,99 @@ var getChartData = function () {
     $(document).trigger(e);
 };
 
-var previousScreen = function () {
-    var e = $.Event('keydown');
+const previousScreen = function () {
+    const e = $.Event('keydown');
     e.keyCode = 37; // arrow-left
     $(document).trigger(e);
 };
 
-var REGULAR_ANSWER_COLOR = '#212529';
-var CORRECT_ANSWER_COLOR = '#2CBB70';
-var WRONG_ANSWER_COLOR = '#D9534F';
+const REGULAR_ANSWER_COLOR = '#212529';
+const CORRECT_ANSWER_COLOR = '#2CBB70';
+const WRONG_ANSWER_COLOR = '#D9534F';
 
 /**
  * A 'regular' answer is an answer that is nor correct, nor incorrect.
  * The check is based on the specific opacity (0.8) and color of those answers.
  */
-var isRegularAnswer = function (answer) {
+const isRegularAnswer = function (answer) {
     return answer.backgroundColor.includes('0.8') &&
         answer.labelColor === REGULAR_ANSWER_COLOR;
 };
 
 /**
  * The check is based on the specific opacity (0.8) and color of correct answers.
  */
-var isCorrectAnswer = function (answer) {
+const isCorrectAnswer = function (answer) {
     return answer.backgroundColor.includes('0.8') &&
         answer.labelColor === CORRECT_ANSWER_COLOR;
 };
 
 /**
  * The check is based on the specific opacity (0.2) and color of incorrect answers.
  */
-var isIncorrectAnswer = function (answer) {
+const isIncorrectAnswer = function (answer) {
     return answer.backgroundColor.includes('0.2') &&
         answer.labelColor === WRONG_ANSWER_COLOR;
 };
 
+const _checkAnswersCount = function (chartData, expectedCount) {
+    if (chartData.length !== expectedCount) {
+        throw new TourError(`Chart data should contain ${expectedCount} records!`);
+    }
+};
+
+const _checkAllZeros = function (chartData) {
+    if (chartData.find(answerData => answerData !== 0).length) {
+        throw new TourError('Chart data should all be 0!');
+    }
+};
+
+const _IS_ASPECT_ANSWER_DATA = {
+    "correct": [isCorrectAnswer, 'should be shown as "correct"!'],
+    "incorrect": [isIncorrectAnswer, 'should be shown as "incorrect"!'],
+    "regular": [isRegularAnswer, 'should not be shown as "correct" or "incorrect"!'],
+    "regularScored": [isRegularAnswer, "correctness shouldn't be shown!"],
+};
+const _INDEX_TO_LABEL = {
+    0: 'First',
+    1: 'Second',
+    2: 'Third',
+    3: 'Fourth',
+};
+
+/**
+ * Check the answers count and aspect.
+ *
+ * @param {object} chartData Object returned by `getChartData`.
+ * @param {{value: number, aspect: "correct" | "incorrect" | "regular" | "regularScored"}[]} expectedAnswersData with:
+ *  `value`: expected answers count (number of user inputs)
+ *  `aspect`: Appearance of the answers count. Use `regularScored` for scored question
+ *    (when the correctness should not be shown) to improve the error message
+ */
+const checkAnswers = function (chartData, expectedAnswersData) {
+    _checkAnswersCount(chartData, expectedAnswersData.length);
+
+    zip(chartData, expectedAnswersData).forEach(([actual, expected], answerIndex) => {
```

Comment:
Maybe "answer" and "expectedAnswer" ? "actual" does not mean much taken out of context.